pygame/math.py

```diff
@@ -0,0 +1,1575 @@
+        if other_x is not None and other_y is not None and other_z is not None:
+            return Vector3(pow_compat(self.vector.x, other_x),
+                           pow_compat(self.vector.y, other_y),
+                           pow_compat(self.vector.z, other_z))
+        else:
+            return NotImplemented
+
+    def __rpow__(self, other):
+        other_x = None
+        other_y = None
+        other_z = None
+        if isinstance(other, Number):
+            other_x = other_y = other_z = other
+        elif isinstance(other, Vector3):
+            other_x = other.x
+            other_y = other.y
+            other_z = other.z
+        if other_x is not None and other_y is not None and other_z is not None:
+            return Vector3(pow_compat(other_x, self.vector.x),
+                           pow_compat(other_y, self.vector.y),
+                           pow_compat(other_z, self.vector.z))
+        else:
+            return NotImplemented
+
+    def __mod__(self, other):
+        if isinstance(other, Number):
+            return Vector3(self.vector.x % other,
+                           self.vector.y % other,
+                           self.vector.z % other)
+        elif isinstance(other, Vector3):
+            return Vector3(self.vector.x % other.x,
+                           self.vector.y % other.y,
+                           self.vector.z % other.z)
+        elif isinstance(other, ElementwiseVector3Proxy):
+            return Vector3(self.vector.x % other.vector.x,
+                           self.vector.y % other.vector.y,
+                           self.vector.z % other.vector.z)
+        return NotImplemented
+
+    def __rmod__(self, other):
+        if isinstance(other, Number):
+            return Vector3(other % self.vector.x,
+                           other % self.vector.y,
+                           other % self.vector.z)
+        elif isinstance(other, Vector3):
+            return Vector3(other.x % self.vector.x,
+                           other.y % self.vector.y,
+                           other.z % self.vector.z)
+        return NotImplemented
+
+    def __eq__(self, other):
+        if isinstance(other, Number):
+            dx = self.vector.x - other
+            dy = self.vector.y - other
+            dz = self.vector.z - other
+        elif isinstance(other, Vector3):
+            dx = self.vector.x - other.x
+            dy = self.vector.y - other.y
+            dz = self.vector.z - other.z
+        elif isinstance(other, ElementwiseVector3Proxy):
+            dx = self.vector.x - other.vector.x
+            dy = self.vector.y - other.vector.y
+            dz = self.vector.z - other.vector.z
+        else:
+            return NotImplemented
+        # Note: comparisons like dx == dx are to check for NaN
+        return (dx == dx and dy == dy and dz == dz and
+                abs(dx) < VECTOR_EPSILON and
+                abs(dy) < VECTOR_EPSILON and
+                abs(dz) < VECTOR_EPSILON)
+
```

Comment:
pygame exposes epsilon as a attribute on Vector2 and Vector3 to control the comparison accuracy

```
>>> from pygame.math import Vector2
>>> s=Vector2(2,2)
>>> s2=s+Vector2(1e-8,1e-8)
>>> s2 == s
True
>>> s2.epsilon = 1e-10
>>> s2 == s
False
>>> s == s2
True
```

We should either match this behaviour, or document the discrepency.